File: 1957_DeleteCharactersToMakeFancyString.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        fancy = []
        last = ''
        count = 0
        for c in s:
            if c==last:
                if count<2:
                    count+=1
                    fancy.append(c)
            else:
                fancy.append(c)
                last=c
                count=1

        return ''.join(fancy)

File: 1957_DeleteCharactersToMakeFancyString.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    string makeFancyString(string s) {
        string ans = s.substr(0,2);
        int n = s.length();
        for (int i=2; i<n; ++i){
            if (s[i]!=s[i-1] || s[i]!=s[i-2])   ans += s[i];
        }        
        return ans;
    }
};